share/completions/magento.fish

```diff
@@ -0,0 +1,720 @@
+	echo hi_IN\t"Hindi (India)"
+	echo hu_HU\t"Hungarian (Hungary)"
+	echo is_IS\t"Icelandic (Iceland)"
+	echo id_ID\t"Indonesian (Indonesia)"
+	echo it_IT\t"Italian (Italy)"
+	echo it_CH\t"Italian (Switzerland)"
+	echo ja_JP\t"Japanese (Japan)"
+	echo km_KH\t"Khmer (Cambodia)"
+	echo ko_KR\t"Korean (South Korea)"
+	echo lo_LA\t"Lao (Laos)"
+	echo lv_LV\t"Latvian (Latvia)"
+	echo lt_LT\t"Lithuanian (Lithuania)"
+	echo mk_MK\t"Macedonian (Macedonia)"
+	echo ms_Latn_MY\t"Malay (Malaysia)"
+	echo mn_Cyrl_MN\t"Mongolian (Mongolia)"
+	echo nb_NO\t"Norwegian Bokmål (Norway)"
+	echo nn_NO\t"Norwegian Nynorsk (Norway)"
+	echo fa_IR\t"Persian (Iran)"
+	echo pl_PL\t"Polish (Poland)"
+	echo pt_BR\t"Portuguese (Brazil)"
+	echo pt_PT\t"Portuguese (Portugal)"
+	echo ro_RO\t"Romanian (Romania)"
+	echo ru_RU\t"Russian (Russia)"
+	echo sr_Cyrl_RS\t"Serbian (Serbia)"
+	echo sk_SK\t"Slovak (Slovakia)"
+	echo sl_SI\t"Slovenian (Slovenia)"
+	echo es_AR\t"Spanish (Argentina)"
+	echo es_CL\t"Spanish (Chile)"
+	echo es_CO\t"Spanish (Colombia)"
+	echo es_CR\t"Spanish (Costa Rica)"
+	echo es_MX\t"Spanish (Mexico)"
+	echo es_PA\t"Spanish (Panama)"
+	echo es_PE\t"Spanish (Peru)"
+	echo es_ES\t"Spanish (Spain)"
+	echo es_VE\t"Spanish (Venezuela)"
+	echo sw_KE\t"Swahili (Kenya)"
+	echo sv_SE\t"Swedish (Sweden)"
+	echo th_TH\t"Thai (Thailand)"
+	echo tr_TR\t"Turkish (Turkey)"
+	echo uk_UA\t"Ukrainian (Ukraine)"
+	echo vi_VN\t"Vietnamese (Vietnam)"
+	echo cy_GB\t"Welsh (United Kingdom)"
+end
+
+
+function __fish_print_magento_source_theme_file_types -d "Shows all available source theme file types"
+  echo less\t"Currently, LESS is the only file type supported"
+end
+
+function __fish_print_magento_deploy_modes -d "Shows all available deploy modes"
+  echo developer\t"Development mode"
+  echo production\t"Production mode"
+end
+
+###
+# Retrieves all existing magento admin users from the mysql database
+###
+function __fish_print_magento_users -d "Shows all existing magento admin users"
+  set -l users (__fish_print_magento_mysql_command 'select username from admin_user;')
+
+  for i in $users
+    echo $i
+  end
+end
+
+function __fish_print_magento_cache_types -d "Shows all available cache types"
+  echo config\t"Configuration"
+  echo layout\t"Layout"
+  echo block_html\t"Block HTML output"
+  echo collections\t"Collections data"
+  echo db_ddl\t"Database schema"
+  echo eav\t"Entity attribute value (EAV)"
+  echo full_page\t"Page cache"
+  echo reflection\t"Reflection"
+  echo translate\t"Translations"
+  echo config_integration\t"Integration configuration"
+  echo config_integration_api\t"Integration API configuration"
+  echo config_webservice\t"Web services configuration"
+end
+
+function __fish_print_magento_verbosity_levels -d "Shows all available verbosity levels"
+  echo 1\t"normal output"
+  echo 2\t"more verbose output"
+  echo 3\t"debug"
+end
+
+function __fish_print_magento_list_formats -d "Shows all available output formats"
+  echo txt\t"Prints as plain text"
```

Comment:
This description (as well as almost all descriptions in this file) has been taken from the help output of the original Magento application. I wanted to stay with the original as closely as possible.